### PR TITLE
refactor: Migrate to Go modules

### PR DESCRIPTION
**Pull request by ondrej-fabry**
- [x] migrate to Go modules
- [x] use vanity import path `go.ligato.io/cn-infra/v2`
- [x] remove deprecated docs directory
- [x] update .md documents
- [x] refactor Makefile
- [x] cleanup .gitignore files
- [x] use `golangci-lint` linter (`gometalinter` has been deprecated)
- [x] optimize all imports

### CODE REVIEW

File: datasync/kvdbsync/local/local_bytes_txn.go [hunk 1]
```diff
@@ -17,10 +17,11 @@ package local
 import (
 	"sync"
```
Comment (rewenset) [inline datasync/kvdbsync/local/local_bytes_txn.go]: extra empty line

File: datasync/restsync/rest_watcher.go [hunk 2]
```diff
@@ -15,14 +15,14 @@
...
+	"go.ligato.io/cn-infra/v2/datasync"
+	"go.ligato.io/cn-infra/v2/logging/logrus"
 
 	"net/http"
 
 	"github.com/gorilla/mux"
-	"github.com/ligato/cn-infra/datasync/syncbase"
 	"github.com/unrolled/render"
+	"go.ligato.io/cn-infra/v2/datasync/syncbase"
 )
```
Comment (rewenset) [inline datasync/restsync/rest_watcher.go]: weird import order

File: db/keyval/etcd/etcd_integration_test.go [hunk 3]
```diff
@@ -19,14 +19,15 @@ import (
...
-	"github.com/ligato/cn-infra/db/keyval/etcd/mocks"
-	"github.com/ligato/cn-infra/logging/logrus"
+	"go.ligato.io/cn-infra/v2/datasync"
+	"go.ligato.io/cn-infra/v2/db/keyval"
+	"go.ligato.io/cn-infra/v2/db/keyval/etcd/mocks"
+	"go.ligato.io/cn-infra/v2/logging/logrus"
+
+	"context"
 
 	"github.com/coreos/etcd/etcdserver/api/v3client"
```
Comment (rewenset) [inline db/keyval/etcd/etcd_integration_test.go]: one more

File: examples/cassandra-lib/client-config.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 endpoints:
   - 127.0.0.1:9042
-op_timeout: 5000000000  # 5 seconds (creating a new keyspace is very slow on travis)
+op_timeout: 5000000000  # 5 seconds (creating a new keyspace is slow on travis)
```
Comment (rewenset) [inline examples/cassandra-lib/client-config.yaml]: it is just "slow" now? :)

Comment (ondrej-fabry) [reply examples/cassandra-lib/client-config.yaml]: This was done to shorten the line and avoid showing warning from inspection.. hehe.

File: Makefile [hunk 5]
```diff
@@ -1,181 +1,203 @@
...
+	cd examples/flags-lib                       && go clean
+	cd examples/kafka-plugin/hash-partitioner   && go clean
+	cd examples/kafka-plugin/manual-partitioner && go clean
+	cd examples/kafka-plugin/post-init-consumer && go clean
+	cd examples/logs-plugin                     && go clean
+	cd examples/redis-plugin                    && go clean
+	cd examples/simple-agent                    && go clean
+	cd examples/statuscheck-plugin              && go clean
+	cd examples/prometheus-plugin               && go clean
+	cd examples/bolt-plugin                     && go clean
```
Comment (rewenset) [inline Makefile]: looks like `examples/cryptodata-plugin` missing

Comment (rewenset) [reply Makefile]: maybe where is a way to `forloop` it

Comment (ondrej-fabry) [reply Makefile]: Actually, we could use `go list` to list all Go programs (`main` packages) under `examples` directory:

```
➤ go list -f '{{if eq .Name "main"}}{{.Dir}}{{end}}' ./examples/...
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/bolt-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cassandra-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/configs-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/consul-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-proto-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/datasync-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/editor
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/election
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/view
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/watcher
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/flags-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-client
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-server
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/asyncproducer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/consumer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/mux
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/syncproducer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/hash-partitioner
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/manual-partitioner
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/post-init-consumer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/basic
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/custom
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/http
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/advanced-scenario
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/basic-scenario
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/templates
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/test-process
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/prometheus-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/airport
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/simple
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/simple-agent
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/statuscheck-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/supervisor-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/01_hello-world
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/02_plugin-deps
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/03_rest-handler
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/04_kv-store
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/06_plugin_lookup
```

And then we can actually pipe this output into `go clean` using `xargs`:

```
➤ go list -f '{{if eq .Name "main"}}{{.Dir}}{{end}}' ./examples/... | xargs go clean -x
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/bolt-plugin
rm -f bolt-plugin bolt-plugin.exe bolt-plugin.test bolt-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cassandra-lib
rm -f cassandra-lib cassandra-lib.exe cassandra-lib.test cassandra-lib.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/configs-plugin
rm -f configs-plugin configs-plugin.exe configs-plugin.test configs-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/consul-lib
rm -f consul-lib consul-lib.exe consul-lib.test consul-lib.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-lib
rm -f cryptodata-lib cryptodata-lib.exe cryptodata-lib.test cryptodata-lib.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-plugin
rm -f cryptodata-plugin cryptodata-plugin.exe cryptodata-plugin.test cryptodata-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-proto-plugin
rm -f cryptodata-proto-plugin cryptodata-proto-plugin.exe cryptodata-proto-plugin.test cryptodata-proto-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/datasync-plugin
rm -f datasync-plugin datasync-plugin.exe datasync-plugin.test datasync-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/editor
rm -f editor editor.exe editor.test editor.test.exe doc doc.exe editor editor.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/election
rm -f election election.exe election.test election.test.exe election election.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/view
rm -f view view.exe view.test view.test.exe doc doc.exe view view.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/watcher
rm -f watcher watcher.exe watcher.test watcher.test.exe doc doc.exe watcher watcher.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/flags-lib
rm -f flags-lib flags-lib.exe flags-lib.test flags-lib.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-client
rm -f grpc-client grpc-client.exe grpc-client.test grpc-client.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-server
rm -f grpc-server grpc-server.exe grpc-server.test grpc-server.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/asyncproducer
rm -f asyncproducer asyncproducer.exe asyncproducer.test asyncproducer.test.exe asyncproducer asyncproducer.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/consumer
rm -f consumer consumer.exe consumer.test consumer.test.exe consumer consumer.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/mux
rm -f mux mux.exe mux.test mux.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/syncproducer
rm -f syncproducer syncproducer.exe syncproducer.test syncproducer.test.exe doc doc.exe syncproducer syncproducer.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/hash-partitioner
rm -f hash-partitioner hash-partitioner.exe hash-partitioner.test hash-partitioner.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/manual-partitioner
rm -f manual-partitioner manual-partitioner.exe manual-partitioner.test manual-partitioner.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/post-init-consumer
rm -f post-init-consumer post-init-consumer.exe post-init-consumer.test post-init-consumer.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/basic
rm -f basic basic.exe basic.test basic.test.exe basic basic.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/custom
rm -f custom custom.exe custom.test custom.test.exe custom custom.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/http
rm -f http http.exe http.test http.test.exe doc doc.exe server server.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-plugin
rm -f logs-plugin logs-plugin.exe logs-plugin.test logs-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/advanced-scenario
rm -f advanced-scenario advanced-scenario.exe advanced-scenario.test advanced-scenario.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/basic-scenario
rm -f basic-scenario basic-scenario.exe basic-scenario.test basic-scenario.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/templates
rm -f templates templates.exe templates.test templates.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/test-process
rm -f test-process test-process.exe test-process.test test-process.test.exe test-process test-process.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/prometheus-plugin
rm -f prometheus-plugin prometheus-plugin.exe prometheus-plugin.test prometheus-plugin.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/airport
rm -f airport airport.exe airport.test airport.test.exe airport airport.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/simple
rm -f simple simple.exe simple.test simple.test.exe doc doc.exe simple simple.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-plugin
rm -f redis-plugin redis-plugin.exe redis-plugin.test redis-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/simple-agent
rm -f simple-agent simple-agent.exe simple-agent.test simple-agent.test.exe agent agent.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/statuscheck-plugin
rm -f statuscheck-plugin statuscheck-plugin.exe statuscheck-plugin.test statuscheck-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/supervisor-plugin
rm -f supervisor-plugin supervisor-plugin.exe supervisor-plugin.test supervisor-plugin.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/01_hello-world
rm -f 01_hello-world 01_hello-world.exe 01_hello-world.test 01_hello-world.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/02_plugin-deps
rm -f 02_plugin-deps 02_plugin-deps.exe 02_plugin-deps.test 02_plugin-deps.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/03_rest-handler
rm -f 03_rest-handler 03_rest-handler.exe 03_rest-handler.test 03_rest-handler.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/04_kv-store
rm -f 04_kv-store 04_kv-store.exe 04_kv-store.test 04_kv-store.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/06_plugin_lookup
rm -f 06_plugin_lookup 06_plugin_lookup.exe 06_plugin_lookup.test 06_plugin_lookup.test.exe main main.exe plugin1 plugin1.exe plugin2 plugin2.exe
```

This seems most universal way to do it.

**Pull request merged**